storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/BlindedBlockKvStoreDatabase.java

```diff
@@ -80,37 +81,50 @@ protected FinalizedUpdaterBlinded finalizedUpdater() {
   @Override
   public List<SignedBeaconBlock> getNonCanonicalBlocksAtSlot(final UInt64 slot) {
     return dao.getBlindedNonCanonicalBlocksAtSlot(slot).stream()
-        .flatMap(block -> getUnblindedBlock(Optional.of(block)).stream())
+        .flatMap(block -> getUnblindedBlock(Optional.of(block), block.getRoot()).stream())
         .collect(Collectors.toList());
   }
 
   @Override
   public Optional<SignedBeaconBlock> getSignedBlock(final Bytes32 root) {
-    return getUnblindedBlock(dao.getBlindedBlock(root));
+    return getUnblindedBlock(dao.getBlindedBlock(root), root);
   }
 
   private Optional<SignedBeaconBlock> getUnblindedBlock(
-      final Optional<SignedBeaconBlock> maybeBlock) {
+      final Optional<SignedBeaconBlock> maybeBlock, final Bytes32 blockRoot) {
     if (maybeBlock.isEmpty()) {
       return maybeBlock;
     }
     final SignedBeaconBlock block = maybeBlock.get();
-    return Optional.of(getUnblindedBlock(block));
+    return Optional.of(getUnblindedBlock(blockRoot, block));
   }
 
-  private SignedBeaconBlock getUnblindedBlock(final SignedBeaconBlock block) {
+  private SignedBeaconBlock getUnblindedBlock(
+      final Bytes32 blockRoot, final SignedBeaconBlock block) {
```

Comment:
nit: I'd probably swap these arguments around so we're consistently using `block, blockRoot`.

Reply:
usually we put index first so i changed it to this order...

Reply:
Either way is fine, I'd just make the two `getUnblindedBlock` methods have it the same way round.